Add CSV "Page" type

divergentdave opened the pull request:
I wrote a new `Page` type to handle CSV files using this pattern over in openstates/openstates#1411. This PR extracts that work so it can live here, upstream.

Let me know if you have any questions or recommendations. Thanks!

jamesturk commented:
awesome, thanks! (didn't even expect others to have found this yet)

I'm still experimenting w/ the API for this, I think it could be more intuitive- definitely lmk if you have any thoughts/suggestions!

jamesturk commented on spatula/base.py:
```diff
@@ -96,3 +96,43 @@ def convert_pdf(self, filename, type):
+    def do_request(self):
+        return self.scraper.get(self.url)
+
+    def scrape_page_items(self, page_type, url=None, **kwargs):
```
it's late so maybe I'm missing something, but aren't these two functions inherited from AbstractPage?

divergentdave replied:
No, `AbstractPage` only has a constructor. I copied these function prototypes from `Page`, which also inherits from `AbstractPage`.

However, looking at this further, I don't think we really need these two methods here. IIUC, `scrape_page_items()` and `scrape_page()` are only used when following a link from one scraped page to another, respectively for pages that will yield multiple objects or return just one. I doubt that many CSVs will merit following links, so I'll take these two methods out now.

jamesturk commented on spatula/base.py:
```diff
@@ -96,3 +96,43 @@ def convert_pdf(self, filename, type):
+        """
+        yield from page_type(self.scraper, url=url, **kwargs).handle_page()
+
+    def scrape_page(self, page_type, url=None, obj=None, **kwargs):
```
""

divergentdave commented:
Okay, this should be ready to go. I'm glad I took a second pass at this, because I botched imports etc. the first time. I've addressed the review comments about extra methods and arguments to DictReader in follow-up commits.

jamesturk commented:
thanks, looks good!
